src/helpers.rs

```diff
@@ -118,6 +119,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.memory.write_bytes(ptr, data.iter().copied())
     }
 
+    /// Call a function: Push the stack frame and pass the arguments.
+    /// For now, arguments must be scalars (so that the caller does not have to know the layout).
+    fn call_function(
+        &mut self,
+        f: ty::Instance<'tcx>,
+        args: &[Scalar<Tag>],
+        dest: Option<PlaceTy<'tcx, Tag>>,
+        stack_pop: StackPopCleanup,
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        // Push frame.
+        let mir = this.load_mir(f.def, None)?;
+        this.push_stack_frame(
+            f,
+            DUMMY_SP, // There is no call site.
+            mir,
+            dest,
+            stack_pop,
+        )?;
+
+        // Initialize arguments.
+        let mut callee_args = this.frame().body.args_iter();
+        for arg in args {
+            let callee_arg = this.local_place(
+                callee_args.next().expect("callee has fewer arguments than expected")
```

Comment:
Compare the lengths of args and callee_args once and then zip the iterators instead of using next + expect... and I just realized args_iter is not ExactSizeIterator even though it totally could be. Carry on

Reply:
I opened https://github.com/rust-lang/rust/issues/66865 for some easy contribs

Reply:
TBH I actually like the way the checking is explicit here, and not indirected through the length. This matches `Allocation::write_bytes`.